### PR TITLE
Adding code for provisioning (wired is validated)

### PR DESCRIPTION
**Pull request by abimishr**
Problem description:  To add the code for device provisioning from the inventory (wired and wireless). Have worked on the review comments as well.
Fix: Added the code for wired and wireless provisioning
Testing information: Tested for wired. Wireless API is having issues in provisioning
Dependancies SDK 2.6.10
Limitations: Unprovision/deletion is not added yet

### CODE REVIEW

File: plugins/modules/provision_intent.py [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+
+        dev_response = self.dnac_apply['exec'](
+            family="devices",
+            function='get_network_device_by_ip',
+            params={"ip_address": self.validated_config[0]["management_ip_address"]}
+        )
+
+        dev_dict = dev_response.get("response")
+        device_family = dev_dict["family"]
+        if device_family == "Wireless Controller":
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:250]: Can you give a new line before if block?

File: plugins/modules/provision_intent.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+            device_type = "wireless"
+        elif device_family in ["Switches and Hubs", "Routers"]:
+            device_type = "wired"
+        return device_type
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:254]: what would be device type if device_family are not above? Can we make default as None or any other value in else part?

File: plugins/modules/provision_intent.py [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+            ):
+                msg = 'Discovery task with id {0} has not completed - Reason: {1}'.format(
+                    task_id, response.get("failureReason"))
+                self.module.fail_json(msg=msg)
+                return False
+
+            if response.get('progress') != 'In Progress':
+                result = True
+                break
+            time.sleep(3)
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:291]: Add a new line before this

File: plugins/modules/provision_intent.py [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:292]: we can remove this new line

File: plugins/modules/provision_intent.py [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+            site = response.get("response")
+            site_additional_info = site[0].get("additionalInfo")
+            for item in site_additional_info:
+                if item["nameSpace"] == "Location":
+                    site_type = item.get("attributes").get("type")
+
+        return site_type
+
+    def get_wired_params(self):
+
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:332]: can we have docstring?

File: plugins/modules/provision_intent.py [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+        wired_params = {
+            "deviceManagementIpAddress": self.validated_config[0]["management_ip_address"],
+            "siteNameHierarchy": self.validated_config[0].get("site_name")
+        }
+
+        return wired_params
+
+    def get_wireless_params(self):
+        wireless_params = [
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:341]: can we have docstring?

File: plugins/modules/provision_intent.py [hunk 7]
```diff
@@ -0,0 +1,522 @@
...
+            before calling the APIs
+        """
+
+        self.want = {}
+        self.want["device_type"] = self.get_dev_type()
+        if self.want["device_type"] == "wired":
+            self.want["prov_params"] = self.get_wired_params()
+        elif self.want["device_type"] == "wireless":
+            self.want["prov_params"] = self.get_wireless_params()
+        self.msg = "Successfully collected all parameters from playbook " + \
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:395]: make something in else.. some log message?

File: plugins/modules/provision_intent.py [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+                )
+
+        elif self.want.get("device_type") == "wireless":
+            response = self.dnac_apply['exec'](
+                family="wireless",
+                function="provision",
+                op_modifies=True,
+                params=self.want["prov_params"],
+            )
+
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:454]: what happens if not wired and wireless

File: plugins/modules/provision_intent.py [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+        Returns:
+            object: An instance of the class with updated results and status
+            based on the processing of differences.
+        Description:
+            The function processes the differences and, depending on the
+            changes required, it may add, update,or resynchronize devices in
+            Cisco DNA Center. The updated results and status are stored in the
+            class instance for further use.
+        """
+        if self.want.get("device_type") == "wired":
```
Comment (madhansansel) [inline plugins/modules/provision_intent.py:415]: we are getting the device_type twice.. Instead can we assign it to variable and check?

device_type = self.want.get("device_type")

Comment (abimishr) [reply plugins/modules/provision_intent.py:415]: Done

File: plugins/modules/provision_intent.py (hunk 8 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:454]: Done

File: plugins/modules/provision_intent.py (hunk 7 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:395]: Done

File: plugins/modules/provision_intent.py (hunk 6 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:341]: Done

File: plugins/modules/provision_intent.py (hunk 5 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:332]: Done

File: plugins/modules/provision_intent.py (hunk 4 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:292]: Done

File: plugins/modules/provision_intent.py (hunk 3 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:291]: Done

File: plugins/modules/provision_intent.py (hunk 2 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:254]: Done

File: plugins/modules/provision_intent.py (hunk 1 above)
Comment (abimishr) [reply plugins/modules/provision_intent.py:250]: Done

**Pull request merged**